File: dsa-one/recursion-and-backtracking/recursion/pow.cpp
#include<iostream>
using namespace std;
int pow(int a,int b){
    if(b==1){
        return a;
    }
    else if(b==0){
        return 1;
    }
    return a * pow(a,b-1);

}
int main(){
    cout<<pow(2,3)<<endl;
    cout<<pow(3,0)<<endl;
    cout<<pow(4,1)<<endl;
}

File: dsa-one/recursion-and-backtracking/recursion/pow.js
const pow = (a,b)=>{
    if(b===1){
        return a;
    }else if(b===0){
        return 0;
    }
    return a * pow(a,b-1);
}
console.log(pow(23,1))
console.log(pow(25,0))
console.log(pow(3,6))